### ISSUE TITLE
Seasonal Automations - Feature

### ISSUE THREAD

**Issue by farberm** [netzero-labs/netzero#49]
I do not see this option to adjust automations based upon seasons. Is this feature coming in an upgrade? It would be nice so that I do not need to remember to adjust things every spring and fall based upon my electric company (Duke Energy Carolinas). 

Another feature would be the Critical Peak Period which is being oimplemented to handle that scenario.

Thanks for a great add

**Comment by zigam** [comment]
Thanks for the suggestion. Is it just the utility rate plan that changes between seasons, or are the other automation changes? Utility rate plans in the Tesla app do have support for seasons currently.

I'll have a look at Critical Peak Period, Netzero would need programmatic access to these periods to be able to act on them.

**Comment by farberm** [comment]
The number of periods change as well as the time for each period. In addition M-F is different from Sat, Sun.. I added an image of that is helpful about the changes between season. I do not know how NetZero app does the programming but adding a date range to the day of the week section would be helpful….
![IMG_0261](https://github.com/user-attachments/assets/ea344df1-0ee7-40cf-beb5-5f4ed7a3ac5b)

Thanks in advance

**Comment by zigam** [comment]
Thanks for the details. Does the Tesla app allow you to configure this rate plan in the app? If so, are you able to use Time-Based Control, or do you use Netzero automations instead to control Powerwall behavior?

**Comment by farberm** [comment]
Yes you can configure a custom rate plan and set the seasons based upon months. Yes I can do time based control but no option for automation throughout the day to change things to minimize grid usage during peak time. That is what I use netzero for.. only issue is that I have to reprogram it for each season. If there was also season section to the day of the week area that would be great so it changes automatically based upon the season/montj

**Comment by zigam** [comment]
Makes sense, thanks for the additional info!

**Comment by arushevich** [comment]
That would be a great feature to add !

**Comment by zigam** [comment]
Automation group labels make this a bit easier now: you can group seasonal automations and pause or resume them as a group.
